packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Providers/ModelFactories/TestData/ModelFactoriesCustomizationTests/CanReplaceModelMethod/SampleNamespaceModelFactory.cs

```diff
@@ -0,0 +1,13 @@
+using Microsoft.Generator.CSharp.Customization;
+using System;
+
+namespace Sample.Models
+{
+    public static partial class SampleNamespaceModelFactory
+    {
+        public static MockInputModel MockInputModel(string prop1 = default, bool? optionalBool = default)
```

Comment:
Can we also add a test case where we have a nullable struct, where the struct is generated by the library?

Reply:
Discussed offline and the scenario this test would exercise is fixed with this PR. I've gone ahead and added a test for that scenario.